demisto_sdk/commands/validate/validators/base_validator.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import (
+    ClassVar,
+    Generic,
+    Iterable,
+    List,
+    Optional,
+    TypeVar,
+    get_args,
+)
+
+from pydantic import BaseModel
+
+from demisto_sdk.commands.content_graph.objects.base_content import (
+    BaseContent,
+    BaseContentMetaclass,
+)
+
+ContentTypes = TypeVar("ContentTypes", bound=BaseContent)
+
+
+class BaseValidator(ABC, BaseModel, Generic[ContentTypes]):
+    error_code: ClassVar[str]
+    description: ClassVar[str]
+    error_message: ClassVar[str]
+    fixing_message: ClassVar[str] = ""
+    is_auto_fixable: ClassVar[bool]
+    related_field: ClassVar[str]
+    expected_git_statuses: ClassVar[Optional[List[str]]] = None
+    graph: ClassVar[bool] = False
+
+    def get_content_types(self):
+        args = get_args(self.__orig_bases__[0])  # type: ignore
+        if isinstance(args[0], (BaseContent, BaseContentMetaclass)):
+            return args
+        return get_args(args[0])
+
+    def should_run(
+        self,
+        content_item: ContentTypes,
+        ignorable_errors: list,
+        support_level_dict: dict,
+    ) -> bool:
+        """check wether to run validation on the given content item or not.
+
+        Args:
+            content_item (BaseContent): The content item to run the validation on.
+            ignorable_errors (list): The list of the errors that can be ignored.
+            support_level_dict (dict): A dict with the lists of validation to run / not run according to the support level.
+
+        Returns:
+            bool: True if the validation should run. Otherwise, return False.
+        """
+        return all(
+            [
+                isinstance(content_item, self.get_content_types()),
+                should_run_according_to_status(
+                    content_item.git_status, self.expected_git_statuses
+                ),
+                not is_error_ignored(
+                    self.error_code, content_item.ignored_errors, ignorable_errors
+                ),
+                not is_support_level_support_validation(
+                    self.error_code, support_level_dict, content_item.support_level
+                ),
+            ]
+        )
+
+    def is_valid(
+        self,
+        content_items: Iterable[ContentTypes],
+        old_content_items: Iterable[Optional[ContentTypes]],
+    ) -> List[ValidationResult]:
+        raise NotImplementedError
+
+    def fix(
+        self, content_item: ContentTypes, old_content_item: ContentTypes
+    ) -> FixingResult:
+        raise NotImplementedError
+
+    class Config:
+        arbitrary_types_allowed = (
+            True  # allows having custom classes for properties in model
+        )
+
+
+class ValidationResult(BaseModel):
+    validator: BaseValidator
+    message: str
+    content_object: BaseContent
+    old_content_object: Optional[BaseContent] = None
```

Comment:
In case of validation that runs again an old format (Rename / Modified), we expect to receive two list - the list of the new file and the list of the old one (before the changes)